stdlib/http/src/main/java/org/ballerinalang/net/http/clientendpoint/CreateSimpleHttpClient.java

```diff
@@ -86,16 +92,41 @@ public void execute(Context context) {
         }
         senderConfiguration.setTLSStoreType(HttpConstants.PKCS_STORE_TYPE);
 
-        populateSenderConfigurationOptions(senderConfiguration, clientEndpointConfig);
+        populateSenderConfigurations(senderConfiguration, clientEndpointConfig);
+        ConnectionManager poolManager;
+        BMap<String, BValue> userDefinedPoolConfig = (BMap<String, BValue>) configBStruct.get(
+                HttpConstants.USER_DEFINED_POOL_CONFIG);
+
+        if (userDefinedPoolConfig == null) {
+            poolManager = getConnectionManager(globalPoolConfig);
+        } else {
+            poolManager = getConnectionManager(userDefinedPoolConfig);
+        }
+
         HttpClientConnector httpClientConnector = httpConnectorFactory
-                .createHttpClientConnector(properties, senderConfiguration);
+                .createHttpClientConnector(properties, senderConfiguration, poolManager);
         BMap<String, BValue> httpClient = BLangConnectorSPIUtil.createBStruct(context.getProgramFile(),
-                                                                                      HTTP_PACKAGE_PATH,
-                                                                                      HTTP_CLIENT, urlString,
-                                                                                      clientEndpointConfig);
+                                                                              HTTP_PACKAGE_PATH,
+                                                                              HTTP_CLIENT, urlString,
+                                                                              clientEndpointConfig);
         httpClient.addNativeData(HttpConstants.HTTP_CLIENT, httpClientConnector);
         httpClient.addNativeData(HttpConstants.CLIENT_ENDPOINT_CONFIG, clientEndpointConfig);
         configBStruct.addNativeData(HttpConstants.HTTP_CLIENT, httpClientConnector);
         context.setReturnValues((httpClient));
     }
+
+    private ConnectionManager getConnectionManager(BMap<String, BValue> poolStruct) {
+        ConnectionManager poolManager = (ConnectionManager) poolStruct.getNativeData(CONNECTION_MANAGER);
+        if (poolManager == null) {
+            synchronized (this) {
+                if (poolStruct.getNativeData(CONNECTION_MANAGER) == null) {
```

Comment:
Shall we check `poolManager == null` instead of checking native data. Getting value from map add some overhead. wdyt?